From 73b6cdca236ce4b516650fa6b8ad369f6b5b6d81 Mon Sep 17 00:00:00 2001
From: Fr0sT-Brutal <fr0st.brutal@gmail.com>
Date: Tue, 1 Mar 2016 17:04:29 +0300
Subject: [PATCH] Add `-ba` switch for silent mode

Implement `-ba` switch which disables log output completely (silent mode). In fact, it was already introduced in sources but without implementation. Hope I did it right.

---
 CPP/7zip/UI/Console/Main.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/CPP/7zip/UI/Console/Main.cpp b/CPP/7zip/UI/Console/Main.cpp
index a632c1b..df86148 100644
--- a/CPP/7zip/UI/Console/Main.cpp
+++ b/CPP/7zip/UI/Console/Main.cpp
@@ -110,6 +110,7 @@ static const char * const kHelpString =
     "  -ax[r[-|0]][m[-|2]][w[-]]{@listfile|!wildcard} : eXclude archives\n"
     "  -ao{a|s|t|u} : set Overwrite mode\n"
     "  -an : disable archive_name field\n"
+    "  -ba : disable log output totally\n"
     "  -bb[0-3] : set output log level\n"
     "  -bd : disable progress indicator\n"
     "  -bs{o|e|p}{0|1|2} : set output stream for output/error/progress line\n"
@@ -1066,7 +1067,7 @@ int Main2(
       uo.SfxModule = kDefaultSfxModule;

     COpenCallbackConsole openCallback;
-    openCallback.Init(g_StdStream, g_ErrStream, percentsStream, options.DisablePercents);
+    openCallback.Init(options.EnableHeaders ? g_StdStream : NULL, g_ErrStream, percentsStream, options.DisablePercents);
 
     #ifndef Z7_NO_CRYPTO
     bool passwordIsDefined =
@@ -1091,7 +1092,7 @@ int Main2(
     callback.StdOutMode = uo.StdOutMode;
     callback.Init(
       // NULL,
-      g_StdStream, g_ErrStream, percentsStream, options.DisablePercents);
+      options.EnableHeaders ? g_StdStream : NULL, g_ErrStream, percentsStream, options.DisablePercents);
 
     CUpdateErrorInfo errorInfo;
 
@@ -1114,7 +1115,8 @@ int Main2(
 
     retCode = WarningsCheck(hresultMain, callback, errorInfo,
         g_StdStream, se,
-        true // options.EnableHeaders
+        //true 
+        options.EnableHeaders
         );
    #endif
   }
